.github/workflows/ci.yml

```diff
@@ -0,0 +1,178 @@
+      ERROR_ON_WARNINGS: ${{ matrix.ERROR_ON_WARNINGS }}
+      RUN_ANALYZER: ${{ matrix.RUN_ANALYZER }}
+      ANDROID_API: ${{ matrix.ANDROID_API }}
+      ANDROID_NDK: ${{ matrix.ANDROID_NDK }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - uses: actions/setup-python@v2
+
+      - name: Installing Linux Dependencies
+        if: ${{ runner.os == 'Linux' && !env['TEST_X86'] }}
+        # workaround: https://github.com/actions/virtual-environments/issues/1536#issuecomment-698537248
+        run: |
+          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key 2>/dev/null | sudo apt-key add -
+          sudo add-apt-repository 'deb http://apt.llvm.org/focal llvm-toolchain-focal-10 main' -y
+          sudo apt update
+          sudo apt install kcov g++-10 valgrind libcurl4 libcurl4-openssl-dev
+          sudo apt install clang-10 clang-tools llvm
+
+      - name: Installing Linux 32-bit Dependencies
+        if: ${{ runner.os == 'Linux' && env['TEST_X86'] }}
+        run: sudo apt update && sudo apt install gcc-multilib g++-multilib
+
+      - name: Installing CodeChecker
+        if: ${{ contains(env['RUN_ANALYZER'], 'code-checker') }}
+        run: |
+          sudo apt install clang-tidy curl doxygen gcc-multilib python3-dev python3-virtualenv
+          git clone https://github.com/Ericsson/CodeChecker.git --depth 1
+          cd CodeChecker
+          BUILD_LOGGER_64_BIT_ONLY=YES make standalone_package
+          echo "::add-path::$PWD/build/CodeChecker/bin"
+
+      - name: Expose llvm PATH for Mac
+        if: ${{ runner.os == 'macOS' }}
+        run: echo "::add-path::/usr/local/opt/llvm/bin/"
+
+      - name: Installing Android SDK Dependencies
+        if: ${{ env['ANDROID_API'] }}
+        run: |
+          export ANDROID_IMAGE="system-images;android-$ANDROID_API;google_apis;x86"
+          echo "Downloading ndk;$ANDROID_NDK and $ANDROID_IMAGE"
+          echo "y" | $ANDROID_HOME/tools/bin/sdkmanager --install \
+            "ndk;$ANDROID_NDK" "$ANDROID_IMAGE" | \
+            grep -v "\[=" || true # suppress the progress bar, so we get meaningful logs
+
+      - name: Starting Android Simulator
+        if: ${{ env['ANDROID_API'] }}
+        run: bash scripts/start-android.sh
+
+      - name: Test
+        shell: bash
+        run: |
+          pip install --upgrade --requirement tests/requirements.txt
+          [ "${{ matrix.CC }}" ] && export CC="${{ matrix.CC }}"
+          [ "${{ matrix.CXX }}" ] && export CXX="${{ matrix.CXX }}"
+          pytest --capture=no --verbose tests
+
+      - name: "Upload to codecov.io"
+        if: ${{ contains(env['RUN_ANALYZER'], 'cov') }}
+        uses: codecov/codecov-action@v1
+        with:
+          directory: coverage
+
+  archive:
+    name: Create Release Archive
+    runs-on: ubuntu-latest
+    needs: [lint, test]
+    if: ${{ needs.test.result == 'success' && github.repository_owner == 'getsentry' }}
```

Comment:
Should we just run this on release branches? Also, do we really need to wait for tests with this?

Reply:
its what we did before, and I think its better to run an extra set of tests, just in case. I remember times when the version bump script did not correctly update the test assertions and caused failures.